Add context menu "Open in Terminal" option

TranceLove opened the pull request:
## Description
Adds option to open specified folder in terminal app.

Supports [TermOne Plus](https://termoneplus.com/), [Termux](https://termux.dev/en/). [Jack Palovich's Android Terminal](https://github.com/jackpal/Android-Terminal-Emulator) is also supported in code, but not thoroughly tested since it's no longer in development.

Expected user to do all the configuration at Terminal app's own side - [like this](https://github.com/termux/termux-tasker#allow-external-apps-property-optional), the most Amaze can do is check for corresponding app permission and prompt user to grant.

#### Issue tracker   
Fixes #2666

#### Automatic tests
- [x] Added test cases
  
#### Manual tests
- [x] Done  
  
- Device: Nexus 4 emulator
- OS: Android 8

#### Build tasks success  
Successfully running following tasks on local:
- [x] `./gradlew assembledebug`
- [x] `./gradlew spotlessCheck`

TranceLove committed:
Add context menu "Open in Terminal" option

Fixes #2666

TranceLove committed:
Add context menu "Open in Terminal" option

Fixes #2666

TranceLove commented:
Pending tasks:

- [ ] Fix codacy complaints

VishalNehra commented:
Pressing on the option 'open in terminal' does nothing on me device
I just have below logs that was showing error level not sure how helpful it would be

`2024-07-18 02:45:05.277  3045-3951  UAH_JNI                 pid-3045                             E  uah_rule_ctrl id = 0x1   ,value = pkgName:com.amaze.filemanager.debug,type:1,activityName:com.amaze.filemanager.debug/com.amaze.filemanager.ui.activities.MainActivity,isFocused:true|pkgName:com.android.launcher,type:1,activityName:com.android.launcher/.Launcher,isFocused:false
`

EmmanuelMess reviewed (changes requested):
For UX, the "open in terminal" button should explain something when clicked, especially if it won't actually work for most users. As it is right now it doesn't appear to even have a toast. I would add at least a small dialog though, especially because "Expected user to do all the configuration at Terminal app's own side" should be explained to the user probably in a popup dialog.

Appart from that, installing on an emulator with API 34, with TermOne Plus, I can't get the button to work, even when the forlder is accessible (and the contents are listable) from TermOne Plus.

TranceLove committed:
Changes per PR feedback

Use dialog instead of toast to notify user that no terminal app can be found.

TranceLove committed:
Changes per PR feedback

Use dialog instead of toast to notify user that no terminal app can be found.

TranceLove commented on app/src/main/java/com/amaze/filemanager/ui/dialogs/OpenFolderInTerminalFragment.kt:
```diff
@@ -0,0 +1,387 @@
+                TERMONE_PLUS -> {
+                    Intent().also {
+                        it.action = "$TERMONE_PLUS.RUN_SCRIPT"
+                        it.setClassName(TERMONE_PLUS, "$ANDROID_TERM.RunScript")
```
Better not tamper with this. Termone plus's RunScript is still `jackpal.androidterm.RunScript`

https://gitlab.com/termapps/termoneplus/-/blob/master/term/src/main/java/jackpal/androidterm/RunScript.java

TranceLove committed:
Changes per PR feedback #2

- Fix top right menu Open in Terminal not working
- Fix permission granting sequence with PermissionsActivity

TranceLove committed:
Changes per PR feedback #2

- Fix top right menu Open in Terminal not working
- Fix permission granting sequence with PermissionsActivity
